app/models/user.rb

```diff
@@ -10,4 +11,17 @@ class User < ActiveRecord::Base
   validates :password, presence: true, length: { minimum: 6 }, on: :create
 
   sluggable_column :email
+
+  def next_slot_in_queue
+    queue_items.count + 1
+  end
+
+  def remove_from_queue!(queue_item)
+    if queue_items.include? queue_item
+      queue_items.each do |i|
```

Comment:
Be careful with non-descriptive variable names. They can make your code harder to follow.